src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultGlobalConfiguration.java

```diff
@@ -34,6 +34,15 @@ public String getKeyVaultURL() {
     public void setKeyVaultURL(String keyVaultURL) {
         this.keyVaultURL = keyVaultURL;
         save();
+        refresh();
+    }
+
+    private void refresh() {
+        ExtensionList<AzureCredentialsProvider> extensionList = Jenkins.get().getExtensionList(AzureCredentialsProvider.class);
```

Comment:
can't you use `ExtensionList.lookupSingleton(AzureCredentialsProvider.class)` ?
it would simplify this code a fair bit